src/halmos/__main__.py

```diff
@@ -730,7 +626,46 @@ def run_target_contract(ctx: ContractContext, ex: Exec, addr: Address) -> list[E
         finally:
             reset(solver)
 
-    return results
+
+def get_frontier(ctx: ContractContext, depth: int) -> Iterable[Exec]:
+    """
+    Retrieves the frontier states at a given depth.
+
+    If the frontier states have already been computed, the cached results are returned.
+    Otherwise, the generator from compute_frontier() is returned.
+    """
+    frontier_states = ctx.frontier_states
+
+    if depth in frontier_states:
+        return frontier_states[depth]
```

Comment:
```suggestion
    if (frontier := frontier_states.get(depth)) is not None:
        return frontier
```

Reply:
(it doesn't really matter in this case, but in general I prefer the pattern of doing a single lookup vs 2 lookups in a dict)

Reply:
https://github.com/a16z/halmos/pull/472/commits/7b9503493e36795034fd37cfce6423dfbd9bd7ba